staging/src/k8s.io/client-go/tools/auth/exec/types_test.go

```diff
@@ -0,0 +1,137 @@
+		execSkippedFieldNames := sets.NewString(
+			// Cluster uses Config to provide its cluster-specific configuration object.
+			"Config",
+		)
+
+		for i := 0; i < execType.NumField(); i++ {
+			execField := execType.Field(i)
+			if execSkippedFieldNames.Has(execField.Name) {
+				continue
+			}
+
+			t.Run(execField.Name, func(t *testing.T) {
+				t.Parallel()
+				clientcmdField, ok := clientcmdType.FieldByName(execField.Name)
+				if !ok {
+					t.Errorf("unknown field (please add field to clientcmdv1.Cluster): '%s'", execField.Name)
+				} else if execField.Type != clientcmdField.Type {
+					t.Errorf(
+						"type mismatch (please update Cluster.%s field type to match clientcmdv1.Cluster.%s field type): %q != %q",
+						execField.Name,
+						clientcmdField.Name,
+						execField.Type,
+						clientcmdField.Type,
+					)
+				} else if execField.Tag != clientcmdField.Tag {
+					t.Errorf(
+						"tag mismatch (please update Cluster.%s tag to match clientcmdv1.Cluster.%s tag): %q != %q",
+						execField.Name,
+						clientcmdField.Name,
+						execField.Tag,
+						clientcmdField.Tag,
+					)
+				}
+			})
+		}
+	})
+
+	t.Run("clientcmd cluster fields match exec cluster fields", func(t *testing.T) {
+		t.Parallel()
+
+		// These are the fields that we don't want to shadow from clientcmdv1.Cluster.
+		clientcmdSkippedFieldNames := sets.NewString(
+			// CA data will be passed via CertificateAuthorityData, so we don't need this field.
+			"CertificateAuthority",
+			// Cluster uses Config to provide its cluster-specific configuration object.
+			"Extensions",
+		)
+
+		for i := 0; i < clientcmdType.NumField(); i++ {
+			clientcmdField := clientcmdType.Field(i)
+			if clientcmdSkippedFieldNames.Has(clientcmdField.Name) {
+				continue
+			}
+
+			t.Run(clientcmdField.Name, func(t *testing.T) {
+				t.Parallel()
+				execField, ok := execType.FieldByName(clientcmdField.Name)
+				if !ok {
+					t.Errorf("unknown field (please add field to Cluster): '%s'", clientcmdField.Name)
+				} else if clientcmdField.Type != execField.Type {
+					t.Errorf(
+						"type mismatch (please update clientcmdv1.Cluster.%s field type to match Cluster.%s field type): %q != %q",
+						clientcmdField.Name,
+						execField.Name,
+						clientcmdField.Type,
+						execField.Type,
+					)
+				} else if clientcmdField.Tag != execField.Tag {
+					t.Errorf(
+						"tag mismatch (please update clientcmdv1.Cluster.%s tag to match Cluster.%s tag): %q != %q",
+						clientcmdField.Name,
+						execField.Name,
+						clientcmdField.Tag,
+						execField.Tag,
+					)
+				}
+			})
+		}
+	})
+}
+
+// TestV1ClusterTypesAreSynced is a TODO so that we remember to write a test similar to
+// TestV1beta1ClusterTypesAreSynced after we promote ExecCredential to GA. It should start failing
+// when someone adds support for the v1 type to this package.
+func TestV1ClusterTypesAreSynced(t *testing.T) {
+	if _, ok := scheme.AllKnownTypes()[schema.GroupVersionKind{
```

Comment:
this is fine, we could also iterate through AllKnownTypes and fail if we find any `clientauthenticationv1beta1.SchemeGroupVersion.Group / ExecCredential` group kinds outside the versions we already expect (v1alpha1 and v1beta1)

Reply:
Makes sense to me. I guess that would protect against stuff like `v1beta2` sneaking in.

Reply:
962c4f64655